crates/ruff_diagnostics/src/fix.rs

```diff
@@ -157,3 +159,43 @@ impl Fix {
         self
     }
 }
+
+/// A collection of [`Edit`] elements to be applied to a source file.
+#[derive(Debug, Clone)]
+pub struct Edits<'a>(Cow<'a, [Edit]>);
+
+impl<'a> Edits<'a> {
```

Comment:
I'm disappointed with what's happening here. The ergonomics of it are fine, but it's so inefficient. `inorder` does a full clone, `retain` does a full clone, etc.